svthalia/ThaliApp
# User authentication using OAuth2

Closes #141

### Summary
This switches our authentication method to using OAuth instead of the previously used "proprietary" token authentication. It also completely rewrites how network requests are executed.

### How to test
Steps to test the changes you made:
1. Login using your local instance
  - To do this you need to define the client id of your application
  - For production and staging these are registered here (if they already exist):
     https://thalia.nu/admin/oauth2_provider/application/
     https://staging.thalia.nu/admin/oauth2_provider/application/
  - This only works when you have this option available: https://github.com/svthalia/concrexit/pull/1353
2. Check every place where we do network requests
3. All works!
4. Now revoke your access token here: https://thalia.nu/admin/oauth2_provider/accesstoken/
5. The app should now request a new access token using the refresh token
6. All still works!

### Todo
- [x] Add documentation to getting started
- [x] Fix CI builds
